tests/test_unknown_schema.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright 2017 Pixar Animation Studios
+#
+# Licensed under the Apache License, Version 2.0 (the "Apache License")
+# with the following modification; you may not use this file except in
+# compliance with the Apache License and the following modification to it:
+# Section 6. Trademarks. is deleted and replaced with:
+#
+# 6. Trademarks. This License does not grant permission to use the trade
+#    names, trademarks, service marks, or product names of the Licensor
+#    and its affiliates, except as required to comply with Section 4(c) of
+#    the License and to reproduce the content of the NOTICE file.
+#
+# You may obtain a copy of the Apache License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the Apache License with the above modification is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied. See the Apache License for the specific
+# language governing permissions and limitations under the Apache License.
+#
+
+import unittest
+
+import opentimelineio as otio
+
+has_undefined_schema = """
+{
+    "OTIO_SCHEMA": "Clip.1",
+    "effects": [],
+    "markers": [],
+    "media_reference": {
+        "OTIO_SCHEMA": "ExternalReference.1",
+        "available_range": {
+            "OTIO_SCHEMA": "TimeRange.1",
+            "duration": {
+                "OTIO_SCHEMA": "RationalTime.1",
+                "rate": 24,
+                "value": 140
+            },
+            "start_time": {
+                "OTIO_SCHEMA": "RationalTime.1",
+                "rate": 24,
+                "value": 91
+            }
+        },
+        "metadata": {
+            "OTIO_SCHEMA": "MyOwnDangSchema.3",
+            "some_data": 895,
+            "howlongami": {
+                "OTIO_SCHEMA": "RationalTime.1",
+                "rate": 30,
+                "value": 100
+            }
+        },
+        "name": null,
+        "target_url": "/usr/tmp/some_media.mov"
+    },
+    "metadata": {},
+    "name": null,
+    "source_range": null
+}
+"""
+
+class UnknownSchemaTests(unittest.TestCase, otio.test_utils.OTIOAssertions):
+    def setUp(self):
+        # make an OTIO data structure containing an undefined schema object
+        self.orig = otio.adapters.otio_json.read_from_string(has_undefined_schema)
+
+    def test_serialize_deserialize(self):
+        serialized = otio.adapters.otio_json.write_to_string(self.orig)
+        test_otio = otio.adapters.otio_json.read_from_string(serialized)
```

Comment:
We also usually do a second test where we serialize/deserialize it again just to make sure there isn't some slightly different state if the object starts in memory vs starts from the deserializer.

Reply:
I'm not sure I follow that.  In this case, self.orig was deserialized from a string constant, so it's in memory.  Then we serialize and deserialize it just to verify that the OTIO is the same, ie: that it preserved the undefined schema object.  I really wanted to check that the "serialized" string is the same string as the original string, but that's brittle.  So the second read_from_string is just to get something that can be compared more reliably.